### PR TITLE
Accomodate changes for KFP 2.2 upgrade

### PR DESCRIPTION
**Pull request by rimolive**
This will include all the changes from the kfp 2.2.0 release

## The issue resolved by this Pull Request:
Resolves #[RHOAIENG-6670](https://issues.redhat.com//browse/RHOAIENG-6670)

## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-684`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/684/head
git checkout -b pullrequest 1bd1e19b7d595100f5b843931d3edf1ac65e37f7
oc new-project opendatahub
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-684"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-684`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-684`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-684`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-684`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-684`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-684`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-684`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-684`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-684`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-684`

File: config/internal/apiserver/default/kfp_launcher_config.yaml.tmpl [hunk 1]
```diff
@@ -10,12 +10,24 @@ data:
...
-        secretName: {{.ObjectStorageConnection.CredentialsSecret.SecretName}}
-        accessKeyKey: {{.ObjectStorageConnection.CredentialsSecret.AccessKey}}
-        secretKeyKey: {{.ObjectStorageConnection.CredentialsSecret.SecretKey}}
+      default:
+        endpoint: {{.ObjectStorageConnection.Endpoint}}
+        {{ if not .ObjectStorageConnection.Secure }}
+        disableSSL:  false
+        {{else}}
+        disableSSL: true
+        {{end}}
```
Comment (HumairAK) [inline config/internal/apiserver/default/kfp_launcher_config.yaml.tmpl:19]: this is the other way around, if it's secure then we `disable:SSL: false`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-684`

**Review by gmfrasca — COMMENTED** [review]
/lgtm

**Pull request merged**